src/query.c

```diff
@@ -385,12 +384,16 @@ SDB_API char *sdb_querys (Sdb *r, char *buf, size_t len, const char *_cmd) {
 	} else if (*cmd == '+' || *cmd == '-') {
 		d = 1;
 		if (!buf) {
-			buf = strdup ("");
+			buf = malloc (len);
```

Comment:
thats not the same, buf needs to be null terminated, use calloc if len matters here